test/lib/base-controller.js

```diff
@@ -322,14 +322,39 @@ describe('lib/base-controller', () => {
           req.sessionModel.reset.should.have.been.calledOnce;
         });
 
+        it('resets the session before callback is called', () => {
+          req.sessionModel.reset.should.have.been.calledBefore(callback);
+        });
+
+      });
+
+      describe('when parent controller getValues errors', () => {
+
+        let error;
+
+        beforeEach(() => {
+          error = 'Parent getValues error.';
+          hmpoFormWizard.Controller.prototype.getValues = sinon.stub().yields(error);
+          controller = new Controller({template: 'foo'});
+          controller.options = {
+            clearSession: true
+          };
+          controller.getValues(req, res, callback);
+        });
+
+        it('immediately fires the callback with the error', () => {
+          callback.should.have.been.calledWith(error);
+          req.sessionModel.reset.should.not.have.been.calledOnce;
```

Comment:
`calledOnce` should only be used to check something is called once, we should be more explicit here and use `called`